src/notion-provider.ts

```diff
@@ -33,82 +50,183 @@ function NotionProvider(this: any, options: NotionProviderOptions) {
       page: {
         cmd: {
           list: {
-	    action: async function(this: any, entize: any, msg: any) {
-	      let limit
-	      if(msg.q){
-                limit = msg.q.limit$ || 100 // The maximum number of results to display per page
-	      }
-	      else{
-	       limit = 100
-	      }
-	      const options = {
-	        method: 'POST',
-		headers: {
-		  'Authorization' : `Bearer ${this.shared.authTok}`,
-		  'Accept': 'application/json',
-		  'Notion-Version': '2022-02-22',
-		  'Content-Type': 'application/json'
-		},
-		body: JSON.stringify({page_size: limit})
-	      };
-	      // see https://developers.notion.com/reference/post-search for usage
-	      let res = await fetch('https://api.notion.com/v1/search', options)
-	      let obj = await res.json()
-	      let list = obj.results.filter(v => v.object == 'page')
-				    .map(v => entize(v))
-	      return list
+            action: async function(this: any, entize: any, msg: any) {
+              let q = msg.q || {}
+
+              const config = {
+                body: { ...q }
+              }
+
+              // see https://developers.notion.com/reference/post-search for usage
+              let res: any = await postJSON(api_endpoint, makeConfig(config))
+
+              let list = res.results.filter((v: any) => v.object == 'page')
+                .map((v: any) => entize(v))
+              return list
+            }
+          },
+
+          load: {
+            action: async function(this: any, entize: any, msg: any) {
+              let id = msg.q.id
+              let res: any
+
+              null == id ? this.fail('invalid_id') : null
+
+              res = await getJSON(`https://api.notion.com/v1/pages/${id}`, makeConfig())
+
+              return entize(res)
+            }
+          },
+
+          save: {
+            action: async function(this: any, entize: any, msg: any) {
+              let q: any = msg.q || {}
+              let ent: any = msg.ent
+              let id: any = ent.id
+              let properties: any = ent.properties || {}
+              let res: any
+
+              (!q.db_id && !id) ? this.fail('invalid_db_id') : null
+
+              const config = null == id
+                ? {
+                    method: 'POST',
+                    body: {
+                      'parent': {
+                        'database_id': q.db_id
+                      },
+                      'properties': { ...properties }
+                    }
+                  }
+                : {
+                    method: 'PATCH',
+                    body: JSON.stringify({
+                      'properties': { ...properties }
+                    })
+
+                  };
+              (null == id)
+                ? ( res = await postJSON('https://api.notion.com/v1/pages',
```

Comment:
this url needs to be an option

Reply:
does the same go for https://github.com/senecajs/seneca-notion-provider/blob/6afe31f117b96e79379f02965c4db8cb0b985824/src/notion-provider.ts#L152 ?

maybe this needs to be something like: `options.api.db_endpoint` & `options.api.page_endpoint` because we already have `api_endpoint` for `list`: https://github.com/senecajs/seneca-notion-provider/blob/6afe31f117b96e79379f02965c4db8cb0b985824/src/notion-provider.ts#L137

`api_endpoint` may be too general here in that case.